lib/split/services/time_based_conversions.rb

```diff
@@ -3,14 +3,15 @@ module Services
     module TimeBasedConversions
 
       def self.within_conversion_time_frame?(user_id, experiment_id)
-        window_of_time_for_conversion = Split.configure.experiments[experiment_id].window_of_time_for_conversion
-        time_of_engagement = Split.redis.get(user_id + "-" + experiment_id)
 
-        (Time.now - time_of_engagement) <= window_of_time_for_conversion
+        window_of_time_for_conversion = Split.configuration.experiments[experiment_id]["window_of_time_for_conversion"]
+        time_of_assignment = Split.redis.get(user_id.to_s + "-" + experiment_id)
+
+        (Time.now - time_of_assignment)/60 <= window_of_time_for_conversion
       end
 
       def self.save_time_that_user_is_assigned(user_id, experiment_name)
-        Split.redis.set(user_id + "-" + experiment_name, Time.now.to_s, ex: 2.months.from_now)
+        Split.redis.set(user_id.to_s + "-" + experiment_name, Time.now.to_s)
```

Comment:
We should have an expiry or cleanup strategy. Quickly checking back at how split manages redis, only objects maintained with the persistence adapter have expiry. And ones outside it (experiment) are cleaned up manually through deletion and would otherwise persist indefinitely.